### DISCUSSION TITLE
Excel Export Action in windows is not working.

### DISCUSSION THREAD

**Discussion by pranab-acharya** [filamentphp/filament#12610] [Help]
### Package

Actions

### Package Version

v3.2

### How can we help you?

I followed all the steps from the documentation for the filament table export but it's not working in windows 11.
![image](https://github.com/filamentphp/filament/assets/152287594/3c19695f-867b-42b0-9405-767e92372e11)
when i am start exporting a table, i got this notification that it's started in the background.

![image](https://github.com/filamentphp/filament/assets/152287594/e1c3be7c-9910-46e4-bdd6-f38bdd3f151d)
but when i check in the queue it's failing, please guide me what's the issue;

**Comment by leandrocfe** [comment]
what is the error?

**Reply by pranab-acharya** [reply]
ExportCsv is failing, so that the table is not exporting, and i want to directly downloading instead of showing in the notification area.

**Reply by leandrocfe** [reply]
Check the failed_jobs table and share the error message

**Reply by pranab-acharya** [reply]
i have uploaded the tables screenshots please check.

**Comment by pranab-acharya** [comment]
1. exports table
![image](https://github.com/filamentphp/filament/assets/152287594/69ad4083-87ee-4a01-95b0-9a5faa0553c8)
2. job batches table
![image](https://github.com/filamentphp/filament/assets/152287594/a2f70549-c396-46f5-bda1-22cd53f85977)
3. failed_jobs table (it's blank)
![image](https://github.com/filamentphp/filament/assets/152287594/a7e54654-6af5-4ed5-8f8f-b23a0d042ced)

**Comment by pranab-acharya** [comment]
The issue was for a column with datatype Enum , commented that column in the Exporter file, now it's working.

**Reply by LucasSouzaa97** [reply]
I had the same problem here but I need to use the enum, did you find any alternative?

**Reply by pranab-acharya** [reply]
yes, you can use the closure to convert the enum to a value or use the label.

**Reply by pranab-acharya** [reply]
can you share the code snippet, and the enum value for reference ?

**Comment by MohammadSalahat** [comment]
**When you try to export an Enum column which is being casted in the Model:**

`    
protected $casts = [
        'status' => RequestStatus::class,
    ];
`

in this case you need to change the format of exporting that colm as the follow:
`
            ExportColumn::make('status')
                ->formatStateUsing(fn (Quote $record) => $record->status->getLabel()),
`

Note: that your Enum Class should have a getLabel function:

`
    public function getLabel(): string
    {
        return __('enums.request_status.'.(string) $this->value);
    }
`
